### DISCUSSION TITLE
Erro:  Publicando pacotes com dependências privadas

### DISCUSSION THREAD

**Discussion by osvaldoribeirodev** [balta-io/forum#118] [Tira Dúvidas] [answered]
Bom dia.
Mesmo colocando a configuração: --store-password-in-clear-text no arquivo .yml, continuo tendo problemas na restauração do pacote.

Run dotnet build --configuration Release
  dotnet build --configuration Release
  shell: /usr/bin/bash -e {0}
  env:
    DOTNET_ROOT: /home/runner/.dotnet
  
MSBuild version 17.3.[2](https://github.com/osvaldoribeirodev/my-second-private-package/actions/runs/3503250801/jobs/5868223447#step:5:2)+5618[4](https://github.com/osvaldoribeirodev/my-second-private-package/actions/runs/3503250801/jobs/5868223447#step:5:4)8881 for .NET
  Determining projects to restore...
/home/runner/work/my-second-private-package/my-second-private-package/MySecondPrivatePackage.csproj : error NU1101: Unable to find package MyCustomPrivatePackage. No packages exist with this id in source(s): github, nuget.org
  Failed to restore /home/runner/work/my-second-private-package/my-second-private-package/MySecondPrivatePackage.csproj (in [5](https://github.com/osvaldoribeirodev/my-second-private-package/actions/runs/3503250801/jobs/5868223447#step:5:5)2[7](https://github.com/osvaldoribeirodev/my-second-private-package/actions/runs/3503250801/jobs/5868223447#step:5:8) ms).

Build FAILED.

/home/runner/work/my-second-private-package/my-second-private-package/MySecondPrivatePackage.csproj : error NU1[10](https://github.com/osvaldoribeirodev/my-second-private-package/actions/runs/3503250801/jobs/5868223447#step:5:11)1: Unable to find package MyCustomPrivatePackage. No packages exist with this id in source(s): github, nuget.org
    0 Warning(s)
    1 Error(s)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @osvaldoribeirodev, como vai?

O erro diz que não encontrou o pacote `MyCustomPrivatePackage`... Se fosse algum problema no Token o erro seria de permissão... você consegue confirmar que o pacote `MyCustomPrivatePackage` existe nos packages da sua empresa?

**Reply by osvaldoribeirodev** [reply]
Bom dia, @andrebaltieri , tudo bem e com vc?

O pacote MyCustomPrivatePackage  existe sim na minha conta e inclusive ele foi criado no curso de Fundamentos Azure. Segui o seu passo a passo e não consigo fazer a Action funcionar. Não sei se o motivo é pelo fato da minha conta ser free.

![image](https://user-images.githubusercontent.com/63430723/203059835-cb83fc98-ebf1-4102-a07e-f3ead7372c41.png)

![image](https://user-images.githubusercontent.com/63430723/203059949-a42d112e-39f5-4ab6-8e7c-31b2618c8dd9.png)

**Reply by andrebaltieri** [reply]
Estranho, a mensagem diz isto:
```
NU1101: Unable to find package MyCustomPrivatePackage. No packages exist with this id in source(s): github, nuget.org
```

Não encontrou o pacote **MyCustomPrivatePackage** nos sources **github** e **nuget.org**.
Imagino que o source **github** seja o que você adicionou seguindo o curso, correto?

Consegue postar o trecho do `yml` que adiciona este source durante o deploy?

**Comment by osvaldoribeirodev** [comment]
Segue o trecho do yml:

- name: Add GitHub Packages Source
      run: dotnet nuget add source https://nuget.pkg.github.com/osvaldoribeirodev/index.json -n github -u osvaldoribeirodev -p ${{secrets.GITHUB_TOKEN}} --store-password-in-clear-text

    - name: Build with dotnet
      run: dotnet build --configuration Release
    
    - name: Create the Package
      run: dotnet pack --configuration Release
    
    - name: Publish
      run: dotnet nuget push "bin/Release/*.nupkg" -k ${{ secrets.GITHUB_TOKEN }} -s https://nuget.pkg.github.com/osvaldoribeirodev/index.json --skip-duplicate
